icu4c/source/common/rbbistbl.cpp

```diff
@@ -85,46 +85,41 @@ const UnicodeString  *RBBISymbolTable::lookup(const UnicodeString& s) const
 
     varRefNode = el->val;
```

Comment:
Please remove the unused variables usetNode, retString, and ("cast off const") This.

Consider moving the remaining variable declarations to their initializations.